### PR TITLE
support threshold based automatic sharding and unsharding of directories

### PR DESCRIPTION
**Pull request by schomatis**
Update UnixFS dependency which now handles HAMT transition internally:
* Remove the deprecated `UseHAMTSharding` option.
* Remove the `addUnixFSChild` intermediary function that was created only to handle HAMT transition. Instead we now add a child interacting with the UnixFS structure directly.

Closes https://github.com/ipfs/go-mfs/issues/87.
Closes https://github.com/ipfs/go-mfs/issues/86.

### CODE REVIEW

**Comment by hsanjuan** [comment]
Hey, a breaking go-unixfs release was made but it fails when mixed with the current go-mfs release because this is not merged, so it would be good to merge and tag.

**Review by hsanjuan — CHANGES_REQUESTED** [review]
Should be bumping go-unixfs version in go.mod?

**Comment by aschmahmann** [comment]
@b5 you should be able to just use the version of go-unixfs used by go-ipfs (v0.2.5) and things should be fine.

Note: go-unixfs v0.2.5 should not have been released, pre v2 go modules are as an ecosystem a bit confusing since things really need to upgrade together which means a release in a dependency shouldn't really happen until most of the important higher ups are ready to be released. When you don't you get issues like the one you're running into.

Meanwhile this is likely blocked on the automatic unixfs unsharding work completing https://github.com/ipfs/go-ipfs/issues/8106 which is underway.

**Comment by b5** [comment]
> @b5 you should be able to just use the version of go-unixfs used by go-ipfs (v0.2.5) and things should be fine.

Sadly that's my problem. I'm filing PRs into projects that use go-unixfs@v0.2.6, with a codebase that has go-ipfs@v0.9.1+ in it's dependencies.

Totally agreed this is just life in go.mod land. I'm just calling attention to this specific conflict 😄, and can work around it if needed

**Review by aschmahmann — APPROVED** [review]

File: go.mod
Comment (aschmahmann) [inline go.mod:15]: TODO: Update to released go-unixfs before merging. cc @guseggert once we do this we're ready to merge here.

**Pull request merged**
